.github/workflows/default.yml

```diff
@@ -77,14 +77,40 @@ jobs:
           gradle-version: ${{ matrix.gradle }}
           arguments: formatKotlin lintKotlin
 
+      - name: Assert fixes
+        run: |
+          cd test-project/build/reports/ktlint
+          grep 'no-empty-class-body' main-format.txt | grep -q 'EmptyClassBodyClass.kt:3:27'
+          grep 'op-spacing' test-format.txt | grep -q 'OpSpacing.kt:5:16'
+
+      - name: Upload reports
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: "test-project-reports-${{ matrix.os }}-${{ matrix.gradle }}-${{ matrix.java }}"
+          path: "${{ github.workspace }}/**/build/reports/ktlint"
+
+  provide-agp-version-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      agp-versions: ${{ steps.build-agp-matrix.outputs.agp-versions }}
+    steps:
+      - id: agp-version-finder
+        uses: usefulness/agp-version-finder-action@v1
+
+      - id: build-agp-matrix
+        run: echo '::set-output name=agp-versions::["${{ steps.agp-version-finder.outputs.latest-stable }}", "${{ steps.agp-version-finder.outputs.latest-alpha }}"]'
+
   integration-tests-android:
     runs-on: ubuntu-latest
+    needs:
+      - provide-agp-version-matrix
     strategy:
       fail-fast: false
       matrix:
         gradle: [ current ]
         java: [ 17 ]
-        agp: [ 7.0.4 ]
+        agp: ${{ fromJSON(needs.provide-agp-version-matrix.outputs.agp-versions) }}
```

Comment:
the `7.2.2` and `7.4.0-alpha10` are calculated dynamically now

<img width="741" alt="image" src="https://user-images.githubusercontent.com/36954793/189500349-659fbe3f-d0e4-48a2-8f04-b7aafe3128f9.png">
<img width="753" alt="image" src="https://user-images.githubusercontent.com/36954793/189500555-6ad94ed3-2956-4df8-ac7c-dc7cfaaa2d74.png">


We'll always test against stable both Gradle and AGP + we'll know early when alpha AGP version breaks things

And also a note: this is my personal action i wrote very recently, for exactly the same purpose (I got annoyed by having manually bumping agp version in CI pipeline)
Related article that inspired me to test early against alpha versions: https://slack.engineering/shadow-jobs/

Reply:
ok

Reply:
But let's be mindful of getting too clever. I always ask the question -- will the setup run a year from now? And will someone know how to fix it if it doesn't?

Reply:
>will the setup run a year from now?

I hope it will, it should 😅

>And will someone know how to fix it if it doesn't?

I personally am not that afraid of trying such things. If it breaks, one can a) ignore the failing check (it isn't required and won't block the development), b) easily revert the commit, c) simply replace the values with hardcoded ones.

I got your point though 👍